### PR TITLE
[NoQA] [Wave8] Workspace List Items

### PR DESCRIPTION
**Pull request by MaciejSWM**
### Details

This PR creates a new component: `WorkspacesListRow.tsx`. 
This PR doesn't render this component anywhere. To render it , go to `WorkspacesListPage.js` and override `return` statement:

Step 1:
Add:
`ownerAccountID: policy.ownerAccountID,`
to `.map((policy) => ({`

Step 2:
```
// import
import React, {useMemo, useRef, useState} from 'react';
import {View} from 'react-native';
import variables from '@styles/variables';
import WorkspacesListRow from './WorkspacesListRow';
// import
    const [layout, setLayout] = useState();
    const containerRef = useRef();

    return (
        <View
            ref={containerRef}
            style={[styles.dFlex, styles.gap3, styles.m5]}
            onLayout={(event) => {
                setLayout(event.nativeEvent.layout.width > variables.extraSmallMobileResponsiveHeightBreakpoint ? CONST.LAYOUT_WIDTH.WIDE : CONST.LAYOUT_WIDTH.NARROW);
            }}
        >
            {_.isEmpty(workspaces) ? (
                <FeatureList
                    menuItems={workspaceFeatures}
                    headline="workspace.emptyWorkspace.title"
                    description="workspace.emptyWorkspace.subtitle"
                />
            ) : (
                _.map(workspaces, (item, index) => (
                    <WorkspacesListRow
                        key={`${item.title}${index}`}
                        title={item.title}
                        workspaceIcon={item.icon}
                        fallbackWorkspaceIcon={item.fallbackIcon}
                        // Real data
                        ownerAccountID={item.ownerAccountID}
                        workspaceType={CONST.POLICY.TYPE.FREE}
                        // Randomized data for testing
                        // ownerAccountID={personalDetails ? Object.keys(personalDetails)[Math.floor(Math.random() * Object.keys(personalDetails).length)] : 1}
                        // workspaceType={[CONST.POLICY.TYPE.CORPORATE, CONST.POLICY.TYPE.TEAM, CONST.POLICY.TYPE.FREE][Math.floor(Math.random() * 3)]}
                        layoutWidth={layout}
                        menuItems={[]}
                    />
                ))
            )}
        </View>
    );
```

Screenshot 2023-12-12 at 22 38 36
Screenshot 2023-12-12 at 22 39 04

### Fixed Issues

$ https://github.com/Expensify/App/issues/32956
PROPOSAL:

### Tests

- Verify that no errors appear in the JS console
- Compare the list items to [Figma](https://www.figma.com/file/Xv3yHJooZcxRGkXbAcTZD5/%23ideal-nav-(Ideal-Nav)?type=design&node-id=568-224981&mode=design&t=1YR4tJFQ5g9W1KDV-4) designs.
- [Web+Desktop] Resize the window to test switching between desktop and mobile layouts
- Three dots menu is not implemented - makes more sense to implement it in the parent component during actual screen creation instead of this PR.

### Offline tests

- Same as `Tests`

### QA Steps

- Same as `Tests`

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/85628901/49c82bf3-1abf-456b-922b-e4ec6afd144e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/85628901/7ded8ef2-e385-4864-969a-06e0b335c09c

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/85628901/a09b64b1-d364-4264-99b5-c9e0b4d897a7

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/85628901/81a84d2f-7990-4085-8b92-71c9c1a4de94

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/85628901/5f9a673d-9189-4015-bcb9-e412a58ac800

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/85628901/9c80a088-1c53-4035-87fb-de2d6e98518a

</details>

### CODE REVIEW

**Comment by MaciejSWM** [comment]
Translations: https://expensify.slack.com/archives/C01GTK53T8Q/p1702073914385499?thread_ts=1702035515.118689&cid=C01GTK53T8Q

**Comment by MaciejSWM** [comment]
@aimane-chnaif ready for review

**Comment by shawnborton** [comment]
Visually this looks pretty good. The main feedback I have is, is that the right icon for the free plan? cc @Expensify/design for thoughts. Perhaps we use something simple like the hot dog stand or something?

**Comment by dannymcclain** [comment]
Yeah it's looking pretty good to me too.

I definitely see what you're saying about the icon choice @shawnborton. When we were making these mocks I don't think I ever really thought about it, but the money rocket doesn't seem like it totally fits with free. I don't think it's the worst thing ever, so I don't think it would be the end of the world to do nothing, but I could definitely see us replacing it with something else.

**Comment by dubielzyk-expensify** [comment]
Could we consider plain icons instead? Feels like the illustration is a bit too complex at that small of a size?

**Comment by trjExpensify** [comment]
👋 @aimane-chnaif putting this one on your radar as a core PR to prioritise reviewing for #wave8 on our roadmap. Will you be able to get through an initial review today? Thanks!

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import useLocalize from '@hooks/useLocalize';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import {AvatarSource} from '@libs/UserUtils';
+import useThemeStyles from '@styles/useThemeStyles';
+import CONST from '@src/CONST';
+import {Policy} from '@src/types/onyx';
+
+type WorkspacesListRowProps = WithCurrentUserPersonalDetailsProps & {
+    /** Name of the workspace */
+    title: Policy['name'];
```
Comment (BartoszGrajdek) [inline src/pages/workspace/WorkspacesListRow.tsx:19]: To keep code in the project consistent I would use `Pick<Policy, 'name'> instead - this is what we have been using util now.

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    /** Account ID of the workspace's owner */
+    ownerAccountID: Policy['ownerAccountID'];
```
Comment (BartoszGrajdek) [inline src/pages/workspace/WorkspacesListRow.tsx:22]: Same here ☝🏻

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    workspaceIcon?: AvatarSource | undefined;
+
+    /** Icon to be used when workspaceIcon is not present */
+    fallbackWorkspaceIcon: AvatarSource | undefined;
+
+    /** Renders the component using big screen layout or small screen layout */
+    isWide: boolean | undefined;
+};
+
+const workspaceTypeIcon = (workspaceType: WorkspacesListRowProps['workspaceType']): React.FC<SvgProps> => {
```
Comment (BartoszGrajdek) [inline src/pages/workspace/WorkspacesListRow.tsx:37]: And here ☝🏻

**Comment by MaciejSWM** [comment]
@dubielzyk-expensify I can swap the icons. Please list them and the corresponding workspace types. If those are new icons then I need the SVGs for them.

**Comment by dubielzyk-expensify** [comment]
Keen on @shawnborton and @dannymcclain 's thoughts first. My mock was a suggestion only

**Comment by shawnborton** [comment]
I don't feel too strongly, I really could go either way. @dannymcclain do you have a preference? 

I guess we kinda use more of those simple illustrations for these sorts of things... mostly I'm just thinking if anything, we should try to be as consistent with the pricing page as possible.

**Review by BartoszGrajdek — APPROVED** [review]

**Comment by MaciejSWM** [comment]
@aimane-chnaif I think we can assume that the icon changes aren't going to block your CR here. If we merge to main without the icon changes I think it will still be benefitial as others are starting to depend on some of the components in PRs.

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import {AvatarSource} from '@libs/UserUtils';
+import CONST from '@src/CONST';
+import {Policy} from '@src/types/onyx';
+
+type WorkspacesListRowProps = WithCurrentUserPersonalDetailsProps & {
+    /** Name of the workspace */
+    title: Policy['name'];
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:19]: can we type `string` directly?

Comment (hayata-suenaga) [reply src/pages/workspace/WorkspacesListRow.tsx:19]: same question goes for other fields (ex. `ownerAccountID`).

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+    /** Account ID of the workspace's owner */
+    ownerAccountID: Policy['ownerAccountID'];
+
+    /** Type of workspace. Type personal is not valid in this context so it's omitted */
+    workspaceType: typeof CONST.POLICY.TYPE.FREE | typeof CONST.POLICY.TYPE.CORPORATE | typeof CONST.POLICY.TYPE.TEAM;
+
+    /** Icon to show next to the workspace name */
+    workspaceIcon?: AvatarSource | undefined;
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:28]: if we have `?`, do we need `undefined`?

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+    /** Icon to be used when workspaceIcon is not present */
+    fallbackWorkspaceIcon: AvatarSource | undefined;
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:31]:
```suggestion
    fallbackWorkspaceIcon?: AvatarSource;
```

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+    /** Renders the component using big screen layout or small screen layout */
+    isWide: boolean | undefined;
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:34]:
```suggestion
    isWide?: boolean;
```

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+        case CONST.POLICY.TYPE.CORPORATE:
+            return Illustrations.HotDogStand;
+        case CONST.POLICY.TYPE.TEAM:
+            return Illustrations.SmallRocket;
+        default:
+            throw new Error(`Don't know which icon to serve for workspace type`);
+    }
+};
+
+function WorkspacesListRow({title, workspaceIcon, fallbackWorkspaceIcon, ownerAccountID, workspaceType, currentUserPersonalDetails, isWide}: WorkspacesListRowProps) {
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:50]: Instead of passing a fallback icon as a prop, can we handle fallback in the component above? We can accept a single icon prop for `WorkspaceListRow`.

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const ownerDetails = PersonalDetailsUtils.getPersonalDetailsByIDs([ownerAccountID], currentUserPersonalDetails.accountID)[0];
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:54]:
```suggestion
    const [ownerDetails] = PersonalDetailsUtils.getPersonalDetailsByIDs([ownerAccountID], currentUserPersonalDetails.accountID);
```

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+                return translate('workspace.type.control');
+            case CONST.POLICY.TYPE.TEAM:
+                return translate('workspace.type.collect');
+            default:
+                throw new Error(`Don't know a friendly workspace name for this workspace type`);
+        }
+    }, [workspaceType, translate]);
+
+    if (isWide === undefined) {
+        return <></>;
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:70]: can we return `null`?

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+                />
+                <Text
+                    numberOfLines={1}
+                    style={[styles.flexGrow1, styles.textStrong]}
+                >
+                    {title}
+                </Text>
+                {!isWide && (
+                    <ThreeDotsMenu
+                        menuItems={[]}
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:92]: we don't have any option items at this point?

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+                        anchorPosition={{top: 0, right: 0}}
+                        iconTooltip={undefined}
+                        icon={undefined}
+                        iconFill={undefined}
+                        iconStyles={undefined}
+                        onIconPress={undefined}
+                        anchorAlignment={undefined}
+                        shouldOverlay={undefined}
+                        shouldSetModalVisibility={undefined}
+                        disabled={undefined}
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:102]: why do we need to explicitly pass `undefined` for these props?

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+                    anchorPosition={{top: 0, right: 0}}
+                    iconStyles={[styles.mr2]}
+                    iconTooltip={undefined}
+                    icon={undefined}
+                    iconFill={undefined}
+                    onIconPress={undefined}
+                    anchorAlignment={undefined}
+                    shouldOverlay={undefined}
+                    shouldSetModalVisibility={undefined}
+                    disabled={undefined}
```
Comment (hayata-suenaga) [inline src/pages/workspace/WorkspacesListRow.tsx:161]: same comment as [this one](https://github.com/Expensify/App/pull/32817/files#r1428560147)

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]

File: src/pages/workspace/WorkspacesListRow.tsx (hunk 8 above)
Comment (MaciejSWM) [reply src/pages/workspace/WorkspacesListRow.tsx:50]: @hayata-suenaga I'm doing this because then I'm passing both those icons to `<Avatar/>`:
```
                <Avatar
                    imageStyles={[styles.alignSelfCenter]}
                    size={CONST.AVATAR_SIZE.DEFAULT}
                    source={workspaceIcon}
                    fallbackIcon={fallbackWorkspaceIcon}
                    name={title}
                    type={CONST.ICON_TYPE_WORKSPACE}
                />
```

@aimane-chnaif any preference if I should go for @hayata-suenaga suggestion or leave it as it is?

File: src/pages/workspace/WorkspacesListRow.tsx (hunk 12 above)
Comment (MaciejSWM) [reply src/pages/workspace/WorkspacesListRow.tsx:102]: This was auto generated as ThreeDotsMenu is a JS file so .tsx didn't know which of those are required. I think I can delete now and it will still work. Testing

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+            case CONST.POLICY.TYPE.CORPORATE:
+                return translate('workspace.type.control');
+            case CONST.POLICY.TYPE.TEAM:
+                return translate('workspace.type.collect');
+            default:
+                throw new Error(`Don't know a friendly workspace name for this workspace type`);
+        }
+    }, [workspaceType, translate]);
+
+    if (isWide === undefined) {
```
[the remark this answers is not in the document]
Comment (MaciejSWM) [reply src/pages/workspace/WorkspacesListRow.tsx:69]: Correct. This `isWide` can in reality be either true or false, but when undefined, we don't assume anything and just return null. I'm open to suggestions on how to improve it.

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+                    shouldOverlay={undefined}
+                    shouldSetModalVisibility={undefined}
+                    disabled={undefined}
+                />
+            )}
+        </View>
+    );
+}
+
+export default withCurrentUserPersonalDetails(WorkspacesListRow);
```
[the remark this answers is not in the document]
Comment (MaciejSWM) [reply src/pages/workspace/WorkspacesListRow.tsx:168]: @aimane-chnaif do you mean this?
```
const personalDetails = usePersonalDetails() || CONST.EMPTY_OBJECT;
    console.log({personalDetails});
```
This stores all personal details, not just currentUser.
image

**Comment by MaciejSWM** [comment]
@aimane-chnaif @hayata-suenaga I would leave menuItems empty for now and keep it for the screen implementation phase. Difficult to predict how to code for it should look like right now without broader context. The calculations for menuItems might mostly be happening in the parent component so that's another reason to leave it for the screen that implements it.

File: src/pages/workspace/WorkspacesListRow.tsx (hunk 12 above)
Comment (hayata-suenaga) [reply src/pages/workspace/WorkspacesListRow.tsx]: please close this conversation when you confirm that the code works 🙇

File: src/pages/workspace/WorkspacesListRow.tsx (hunk 8 above)
Comment (hayata-suenaga) [reply src/pages/workspace/WorkspacesListRow.tsx]: Ah, I didn't check the interface of the Avatar component. Let's leave it as is.

**Comment by hayata-suenaga** [comment]
can we still fix this TS error in CI?

```
Error: src/pages/workspace/WorkspacesListRow.tsx(16,9): error TS2305: Module '"@src/types/onyx/OnyxCommon"' has no exported member 'MenuItem'.
```

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+    workspaceType: typeof CONST.POLICY.TYPE.FREE | typeof CONST.POLICY.TYPE.CORPORATE | typeof CONST.POLICY.TYPE.TEAM;
+
+    /** Icon to show next to the workspace name */
+    workspaceIcon?: AvatarSource;
+
+    /** Icon to be used when workspaceIcon is not present */
+    fallbackWorkspaceIcon?: AvatarSource;
+
+    /** Items for the three dots menu */
+    menuItems: MenuItem[];
```
Comment (blazejkustra) [inline src/pages/workspace/WorkspacesListRow.tsx:35]: You can adjust this type once `MenuItem` [PR](https://github.com/Expensify/App/pull/31141) is merged

Comment (MaciejSWM) [reply src/pages/workspace/WorkspacesListRow.tsx]: Updated

File: src/libs/PersonalDetailsUtils.ts [hunk 17]
```diff
@@ -20,7 +20,11 @@ Onyx.connect({
  * @param [defaultValue] optional default display name value
  */
 function getDisplayNameOrDefault(displayName?: string, defaultValue = ''): string {
-    return displayName ?? defaultValue ?? Localize.translateLocal('common.hidden');
+    // Not using nullish coalescing as it differs from OR for strings. For example:
+    // '' || 'A' === 'A'
+    // '' ?? 'A' === ''
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    return displayName || defaultValue || Localize.translateLocal('common.hidden');
```
[the remark this answers is not in the document]
Comment (MaciejSWM) [reply src/libs/PersonalDetailsUtils.ts:27]: Umm, I'm not sure if it fixes any known bugs and definitely I'm hoping that it doesn't create any.

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+            default:
+                throw new Error(`Don't know a friendly workspace name for this workspace type`);
+        }
+    }, [workspaceType, translate]);
+
+    if (renderLayout === CONST.LAYOUT.NONE) {
+        // To prevent layout from jumping or rendering for a split second, when
+        // isWide is undefined we don't assume anything and simply return null.
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (MaciejSWM) [reply src/pages/workspace/WorkspacesListRow.tsx:87]: `useWindowDimensions` takes width of the whole window. While `WorkspacesListRow` might be either rendered inside of RHP or in the central pane, so we need to know the width of it's direct parent component to apply proper styles.

File: src/CONST.ts [hunk 19]
```diff
@@ -1291,6 +1291,12 @@ const CONST = {
         USE_EXPENSIFY_FEES: 'use.expensify.com/fees',
     },
 
+    LAYOUT: {
```
[the remark this answers is not in the document]
Comment (MaciejSWM) [reply src/CONST.ts:1294]: `LAYOUT_WIDTH`? Then it's `LAYOUT_WIDTH.WIDE` and `LAYOUT_WIDTH.NARROW`. I aimed at this being general, didn't want it to be only tied to a single place in the app.

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 20]
```diff
@@ -0,0 +1,170 @@
...
+
+    /** Icon to be used when workspaceIcon is not present */
+    fallbackWorkspaceIcon?: AvatarSource;
+
+    /** Items for the three dots menu */
+    menuItems: MenuItemProps[];
+
+    /** Renders the component using big screen layout or small screen layout. When renderLayout === WorkspaceListRowLayout.NONE,
+     * component will return null to prevent layout from jumping on initial render and when parent width changes. */
+    renderLayout?: ValueOf<typeof CONST.LAYOUT>;
```
[the remark this answers is not in the document]
Comment (MaciejSWM) [reply src/pages/workspace/WorkspacesListRow.tsx:39]: `layoutType`?

[the remark this answers is not in the document]
Comment (MaciejSWM) [reply src/pages/workspace/WorkspacesListRow.tsx:39]: Or maybe `layoutWidth`

**Comment by MaciejSWM** [comment]
@dubielzyk-expensify @dannymcclain @shawnborton 

image

image

**Comment by dannymcclain** [comment]
@MaciejSWM Those look good to me!

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 21]
```diff
@@ -0,0 +1,170 @@
...
+                        style={[styles.textMicro, styles.textSupporting]}
+                    >
+                        {ownerDetails.login}
+                    </Text>
+                </View>
+            </View>
+            <View style={[styles.flexRow, isWide && styles.flex1, styles.gap2, isNarrow && styles.mr5, styles.alignItemsCenter]}>
+                <Icon
+                    src={workspaceTypeIcon(workspaceType)}
+                    width={34}
```
Comment (mountiny) [inline src/pages/workspace/WorkspacesListRow.tsx:140]: Should we make this a const?

**Review by mountiny — DISMISSED** [review]
Looking great, one comment!

All yours @hayata-suenaga

**Comment by dubielzyk-expensify** [comment]
Looks good to me too!

**Comment by MaciejSWM** [comment]
@aimane-chnaif uff you found something really strange:

`ThreeDotsMenu/index.js`:
```
    /** The anchor position of the menu */
    anchorPosition: PropTypes.shape({
        top: PropTypes.number,
        right: PropTypes.number,
        bottom: PropTypes.number,
        left: PropTypes.number,
    }).isRequired,
```

`PopoverMenu/popoverMenuPropTypes.js`:
```
    /** The anchor position of the CreateMenu popover */
    anchorPosition: PropTypes.shape({
        top: PropTypes.number,
        right: PropTypes.number,
        bottom: PropTypes.number,
        left: PropTypes.number,
    }).isRequired,
```

but when `popoverMenuPropTypes.js` is used in `PopoverMenu/index.js` it gets overridden:
```
    /** The horizontal and vertical anchors points for the popover */
    anchorPosition: PropTypes.shape({
        horizontal: PropTypes.number.isRequired,
        vertical: PropTypes.number.isRequired,
    }).isRequired,
```

once overridden, it then is passed to `PopoverWithMeasuredContent.js` where it's being used like this:
```
    // All popover props except:
    // 1) anchorPosition (which is overridden for this component)
    // 2) windowDimensionsPropTypes - we exclude them because we use useWindowDimensions hook instead
    ..._.omit(popoverPropTypes, ['anchorPosition', ..._.keys(windowDimensionsPropTypes)]),

    /** How the popover should be aligned. The value you passed will is the part of the component that will be aligned to the
     * anchorPosition. ie: vertical:top means the top of the menu will be positioned in the anchorPosition */
    anchorAlignment: PropTypes.shape({
        horizontal: PropTypes.oneOf(_.values(CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL)),
        vertical: PropTypes.oneOf(_.values(CONST.MODAL.ANCHOR_ORIGIN_VERTICAL)),
    }),
```

**Comment by MaciejSWM** [comment]
In the end feels like this is the correct type in `Modal/types.ts`:
```
type PopoverAnchorPosition = {
    top?: number;
    right?: number;
    bottom?: number;
    left?: number;
};
```

**Comment by MaciejSWM** [comment]
Adjusting, so that it's `top/right/bottom/left` instead of `horizontal/vertical` for the `anchorPosition` prop.

**Comment by MaciejSWM** [comment]
@aimane-chnaif what is here to fix?
image

In the app feels like we use the popover above top of the item mostly:
image
image
image


We only show the popover below the item if the popover doesn't have space above the item, or so it feels to me.
image

**Comment by blazejkustra** [comment]
Typecheck is failing, will look into it

**Comment by blazejkustra** [comment]
@aimane-chnaif Should be fine now, resolved conflicts and fixed failing typecheck

**Comment by mountiny** [comment]
@aimane-chnaif Can you look into this one after the chat switching performance PR?

**Comment by mountiny** [comment]
> Let's consider anchorPosition issue out of scope (as happening on main) and fix it in separate PR

Yep if its happening in main lets keep it out of this PR

**Review by aimane-chnaif — APPROVED** [review]

**Comment by mountiny** [comment]
@aimane-chnaif 
>Let's create new GH to fix this

Can you summarize the issue and post in expensify-bugs please in the problem/ possible solution format? I can create the issue based on that

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 22]
```diff
@@ -0,0 +1,175 @@
...
+            {isWide && (
+                <ThreeDotsMenu
+                    menuItems={menuItems}
+                    anchorPosition={{top: 0, right: 0}}
+                    iconStyles={[styles.mr2]}
+                />
+            )}
+        </View>
+    );
+}
```
Comment (mountiny) [inline src/pages/workspace/WorkspacesListRow.tsx:173]: are we not adding a displayName for these?

**Review by mountiny — APPROVED** [review]
Appreciated the review @aimane-chnaif and hard work on this one @MaciejSWM! thanks!

**Comment by mountiny** [comment]
Given this is not touching any exsiting flow in the app I am going to merge this now I think Hayata might be out for the holidays too

File: src/pages/workspace/WorkspacesListRow.tsx [hunk 23]
```diff
@@ -0,0 +1,177 @@
...
+                                numberOfLines={1}
+                                style={[styles.labelStrong]}
+                            >
+                                {PersonalDetailsUtils.getDisplayNameOrDefault(ownerDetails.displayName)}
+                            </Text>
+                            <Text
+                                numberOfLines={1}
+                                style={[styles.textMicro, styles.textSupporting]}
+                            >
+                                {ownerDetails.login}
```
Comment (ikevin127) [inline src/pages/workspace/WorkspacesListRow.tsx:136]: `ownerDetails.login` was missing the `Str.removeSMSDomain()` wrapper for cases like user sign up with phone number which without the wrapper is rendering SMS domain `+13237069846@expensify.com` which led to this issue:
- https://github.com/Expensify/App/issues/58864

we fixed it by adding the the `Str.removeSMSDomain()` wrapper around `ownerDetails.login`.
